accord-core/src/main/java/accord/coordinate/tracking/FastPathTracker.java

```diff
@@ -116,7 +116,7 @@ public final ShardOutcome<? super FastPathTracker> onDelayed(@Nonnull Node.Id fr
                 ++fastPathDelayed;
 
                 if (isFastPathDelayed() && hasReachedQuorum())
-                    return complete(Success);
+                    return complete(hasMetMediumPathCriteria() ? NewMediumPathSuccess : Success);
```

Comment:
this logic is duplicated in `accord.coordinate.tracking.FastPathTracker.FastPathShardTracker#quorumIfHasRejectedFastPath`

can we cleanup so we don't have it scattered?

Reply:
What about `accord.coordinate.tracking.FastPathTracker.FastPathShardTracker#onFailure`?  It had the same logic but doesn't include `NewMediumPathSuccess`, why is that?

Reply:
> why is that

I think an oversight; have fixed. Have also factored out the logic to do this as requested.